python/foxglove-sdk/ts/widget.ts

```diff
@@ -0,0 +1,74 @@
+import type { RenderProps } from "@anywidget/types";
+import { FoxgloveViewer, type DataSource, type OpaqueLayoutData } from "@foxglove/embed";
+
+/* Specifies attributes defined with traitlets in ../src/foxglove_notebook/__init__.py */
+interface WidgetModel {
+  width: string;
+  height: string;
+  src: string;
+  orgSlug?: string;
+  data?: DataView<ArrayBuffer>;
+  layout?: OpaqueLayoutData;
```

Comment:
how do we decide which of these are optional and which are not? I don't think I see a difference in how they are defined in python